src/drivers/dw/dma.c

```diff
@@ -639,13 +639,19 @@ static int dw_dma_stop(struct dma *dma, int channel)
 
 	spin_lock_irq(&dma->lock, flags);
 
-	ret = poll_for_register_delay(dma_base(dma) + DW_DMA_CHAN_EN,
-				      CHAN_MASK(channel), 0,
+	dw_write(dma, DW_CFG_LOW(channel), chan->cfg_lo | DW_CFG_CH_SUSPEND |
+		 DW_CFG_CH_DRAIN);
```

Comment:
not sure we need to OR with .cfg_lo above, but it shouldn't hurt either.

Reply:
@lyakh This means we need to keep the other bits to the original status as we only changes few bits